tests/IntegrationTests/EndToEndTests/E2E_UploadFileToMeshStorage.cs

```diff
@@ -0,0 +1,85 @@
+
+namespace Tests.Integration.EndToEndTests
+{
+    [TestClass]
+    [TestCategory("Integration")]
+    public class E2E_UploadFileToMeshStorage : BaseIntegrationTest
+    {
+        private BlobStorageHelper _blobStorageHelper;
+        private string _blobContainerName;
+        private string _episodeFilePath;
+        private string _episodeFileName;
+        private string _participantFilePath;
+        private string _participantFileName;
+        private ILogger<E2E_UploadFileToMeshStorage> _logger;
+        private MeshMailboxHelper _meshMailboxHelper;
+
+        [TestInitialize]
+        public async Task TestInitialize()
+        {
+            _blobStorageHelper = ServiceProvider.GetService<BlobStorageHelper>();
+            _blobContainerName = AppSettings.BlobContainerName;
+            _logger = ServiceProvider.GetService<ILogger<E2E_UploadFileToMeshStorage>>();
+            _meshMailboxHelper = new MeshMailboxHelper(AppSettings.Endpoints.MeshSandboxOutput, AppSettings, HttpClient);
+
+            _episodeFilePath = AppSettings.FilePaths.LocalEpisodesCSVFile;
+            _participantFilePath = AppSettings.FilePaths.LocalSubjectsCSVFile;
+
+            var episodeFileNameIndex = _episodeFilePath.LastIndexOf('/');
+            _episodeFileName = _episodeFilePath.Substring(episodeFileNameIndex+1);
+
+            var participantFileNameIndex = _participantFilePath.LastIndexOf('/');
+            _participantFileName = _participantFilePath.Substring(participantFileNameIndex+1);
+
+            // Ensure all config is set
+            AssertAllConfigurations();
+
+            // Get the direct filepath to the csv test file and ensure it exists
+            var localFilePath = Path.Combine(AppContext.BaseDirectory, AppSettings.FilePaths.LocalEpisodesCSVFile);
+            Assert.IsTrue(localFilePath.Length > 0);
+        }
+
+        [TestMethod]
+        public async Task EndToEnd_UploadEpisodeFileToMeshAndWaitForIngestAndCheckItHasReachedBlobStorage()
+        {
+            var localFilePath = Path.Combine(AppContext.BaseDirectory, _episodeFilePath);
+            await UploadFileToMeshMailboxAsync(_episodeFileName, localFilePath);
+            //In an integrated environment, the default mesh interval value is 5m, which is why this wait period is so long
+            Thread.Sleep(AppSettings.MeshSettings.intervalInMs);
+            Assert.IsTrue(await GetFileFromBlobAsync(_episodeFilePath));
+        }
+
+        [TestMethod]
+        public async Task EndToEnd_UploadParticipantFileToMeshAndWaitForIngestAndCheckItHasReachedBlobStorage()
+        {
+            var localFilePath = Path.Combine(AppContext.BaseDirectory, _participantFilePath);
+            await UploadFileToMeshMailboxAsync(_participantFileName, localFilePath);
+            //In an integrated environment, the default mesh interval value is 5m, which is why this wait period is so long
+            Thread.Sleep(AppSettings.MeshSettings.intervalInMs);
```

Comment:
Wonder if we can replace Thread.Sleep with a more efficient waiting mechanism such as polling with timeouts and something like `await Task.Delay(5000)` I can see why you used Thread.Sleep in this scenario, just concerned it could cause a bit of flakiness in the tests down the line.

Reply:
Resolved using suggestion